### PR TITLE
[MOBILE-2293] Ensure Sample App includes Opt-In dialog, update documentation

### PR DESCRIPTION
**Pull request by sstasiuk**
* Documentation updated
* Sample alert images added
* Added sample alert to project 

Link to branch with updated readme: https://github.com/sewichi/ios-placed-sdk/blob/sample-app-includes-opt-in-dialog/README.md

screen shot 2018-05-10 at 10 03 10 am

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,18 +19,36 @@ If you are not using CocoaPods, you can integrate the Placed.framework and Place
...
+    - Buttons to “Accept” or “Cancel” 
+
+
+1. *Prompt for Background Location Permission*  
+Under the “Required background modes” key in your app’s main plist file, you will need to add:  
+    - **App registers for location updates**   
+    - (for iOS 11) **NSLocationAlwaysAndWhenInUseUsageDescription**
+    - (for iOS 11) **NSLocationWhenInUseUsageDescription**  
+    - (for iOS 11) **NSLocationAlwaysUsageDescription**  
+    - (for iOS 11) **NSLocationUsageDescription**
```
Comment (nickneiman) [inline README.md]: Are these all for ios 11? Don't we use some of these pre 11?

File: README.md [hunk 2]
```diff
@@ -19,18 +19,36 @@ If you are not using CocoaPods, you can integrate the Placed.framework and Place
...
+
+<section>
+	<ul>
+    <img src="./Images/aff-iOS-optin-dialogue1.jpg" width=30%>
+    <img src="./Images/aff-iOS-optin-dialogue2.jpg" width=30%>
+    <img src="./Images/aff-iOS-optin-full1.jpg" width=30%>
+    <img src="./Images/aff-iOS-optin-full2.jpg" width=30%>
+    <img src="./Images/aff-iOS-optin-full3.jpg" width=30%>
+  </ul>
+</section>
```
Comment (nickneiman) [inline README.md]: Does this gallery work in github? I don't see it when I view this file.

File: SampleApp/SampleApp/SampleAlertViewController.m [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+- (IBAction)showTermsOfService:(id)sender {
+    NSURL *url = [[NSURL alloc] initWithString:@"https://www.placed.com/terms-of-service"];
+    [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:nil];
+}
+
+- (IBAction)showPrivacyPolicy:(id)sender {
+    NSURL *url = [[NSURL alloc] initWithString:@"https://www.placed.com/privacy-policy"];
+    [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:nil];
```
Comment (nickneiman) [inline SampleApp/SampleApp/SampleAlertViewController.m]: Our fearless leader prefers dot syntax.

File: SampleApp/SampleApp/ViewController.m [hunk 4]
```diff
@@ -20,6 +21,11 @@ - (void)viewDidLoad {
...
 }
 
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    [self presentViewController:[SampleAlertViewController new] animated:YES completion:nil];
+}
+
 - (IBAction)registerUser:(id)sender {
     // "Register User" button clicked
     [PlacedAgent registerUser];
```
Comment (nickneiman) [inline SampleApp/SampleApp/ViewController.m]: We need to make sure the user has accepted the SampleAlertViewController before registering the user right?

File: SampleApp/SampleApp/SampleAlertViewController.m (hunk 3 above)
Comment (dgranahan) [reply SampleApp/SampleApp/SampleAlertViewController.m]: 🙏

File: README.md (hunk 2 above)
Comment (sstasiuk) [reply README.md]: Well... great, it works only in chrome (at least for me)

File: README.md (hunk 1 above)
Comment (sstasiuk) [reply README.md]: I didn't come up with this text, don't know, thought it was already discussed.

File: README.md (hunk 2 above)
Comment (dgranahan) [reply README.md]: I can't see it looking here: https://github.com/sewichi/ios-placed-sdk/blob/d821dc3be115927a2b0c0f76de2c04ea5057b5e7/README.md

Comment (sstasiuk) [reply README.md]: Kyle just told me that he actually wanted to have images as a link, so this change is not applicable anymore.

File: SampleApp/SampleApp/ViewController.m [hunk 5]
```diff
@@ -22,7 +23,12 @@ - (void)viewDidLoad {
 
 - (IBAction)registerUser:(id)sender {
     // "Register User" button clicked
-    [PlacedAgent registerUser];
+    if ([NSUserDefaults.standardUserDefaults boolForKey:kTermsAndPolicyAcceptedKey]) {
+        [PlacedAgent registerUser];
+    }
+    else {
```
Comment (dgranahan) [inline SampleApp/SampleApp/ViewController.m]: Nitpick: `} else {`

**Comment by dgranahan** [comment]
@sstasiuk just a heads up, I believe we'll need to update some of the SDK method calls (e.g. removing `startTracking()`) when @nickneiman's changes are published to `sewichi/ios-placed-sdk`.

**Comment by sstasiuk** [comment]
> @sstasiuk just a heads up, I believe we'll need to update some of the SDK method calls (e.g. removing startTracking()) when @nickneiman's changes are published to sewichi/ios-placed-sdk.

It's for Android app not for iOS right?

**Comment by nickneiman** [comment]
Both changed a little. iOS changed more. There is not `startTracking` call anymore.

File: README.md (hunk 1 above)
Comment (dgranahan) [reply README.md]: I would vote to remove the "iOS 11" text because they'll be needed for iOS 12 and so on.

File: README.md [hunk 6]
```diff
@@ -19,18 +19,26 @@ If you are not using CocoaPods, you can integrate the Placed.framework and Place
...
+1. *Prompt for Background Location Permission*  
+Under the “Required background modes” key in your app’s main plist file, you will need to add:  
+    - **App registers for location updates**   
+    - (for iOS 11) **NSLocationAlwaysAndWhenInUseUsageDescription**
+    - (for iOS 11) **NSLocationWhenInUseUsageDescription**  
+    - (for iOS 11) **NSLocationAlwaysUsageDescription**  
+    - (for iOS 11) **NSLocationUsageDescription**  
 
-**These permissions are very important to the Placed SDK. If your app does not currently use background location permission, please contact a Placed representative.**
+For an example of the opt-in dialog and location permission prompt, please refer to the sample app. We have also provided a [gallery for inspiration](https://github.com/placed/ios-placed-sdk/blob/master/gallery) on how you can better integrate the opt-in experience into your app.
```
Comment (dgranahan) [inline README.md]: Relative link?

**Review by dgranahan — CHANGES_REQUESTED** [review]
Can we remove image_0.png to avoid confusion?
Also can we remove references to `startTracking` from the README?

**Review by dgranahan — APPROVED** [review]

**Pull request merged**